components/Users/Memberships/CancelMembership.js

```diff
@@ -91,39 +102,55 @@ export default class CancelPledge extends Component {
         cancellationType
       } = this.state
 
-      return mutation({
-        variables: {
-          membershipId: this.props.membership.id,
-          immediately,
-          suppressConfirmation,
-          suppressWinback,
-          details: {
-            reason,
-            type: cancellationType
+      if (this.props.cancellation) {
+        return mutation({
+          variables: {
+            cancellationId: this.props.cancellation.id,
+            details: {
+              reason,
+              type: cancellationType,
+              suppressConfirmation,
+              suppressWinback,
+            }
           }
-        }
-      }).then(() =>
-        this.setState(() => ({ isOpen: false }))
-      )
+        }).then(() =>
+          this.setState(() => ({ isOpen: false }))
+        )
+      } else {
+        return mutation({
+          variables: {
+            membershipId: this.props.membership.id,
+            immediately,
+            details: {
+              reason,
+              type: cancellationType,
+              suppressConfirmation,
+              suppressWinback,
+            }
+          }
+        }).then(() =>
+          this.setState(() => ({ isOpen: false }))
+        )
+      }
     }
   }
 
   render() {
     const { isOpen, reason, cancellationType } = this.state
-    const { refetchQueries, membership } = this.props
+    const { refetchQueries, membership, cancellation } = this.props
     return (
       <Fragment>
         <TextButton
           onClick={() => {
             this.setState({ isOpen: true })
           }}
         >
-          künden
+          { cancellation ? 'editieren' : 'künden' }
         </TextButton>
 
         {isOpen && (
           <Mutation
-            mutation={CANCEL_MEMBERSHIP}
+            mutation={cancellation ? UPDATE_CANCELLATION : CANCEL_MEMBERSHIP}
```

Comment:
Optional:
Looking at this line and others below, I personally feel like "too many cancellations" and would personally introduce `const isCancelled = !!cancellation` for readability, so stuff would read like:
`mutation={isCancelled ? UPDATE_CANCELLATION : CANCEL_MEMBERSHIP}`